source/particle/interpolator/quadratic_least_squares.cc

```diff
@@ -155,39 +250,398 @@ namespace aspect
+            Point<dim> relative_support_point_location = mapping.transform_real_to_unit_cell(found_cell, *itr);
+            for (unsigned int d = 0; d < dim; ++d)
+              relative_support_point_location[d] -= unit_offset;
             for (unsigned int property_index = 0; property_index < n_particle_properties; ++property_index)
               {
-                double interpolated_value = c[property_index][0] +
-                                            c[property_index][1] * relative_support_point_location[0] +
-                                            c[property_index][2] * relative_support_point_location[1];
-                if (dim == 2)
-                  {
-                    interpolated_value += c[property_index][3] * relative_support_point_location[0] * relative_support_point_location[1] +
-                                          c[property_index][4] * relative_support_point_location[0] * relative_support_point_location[0] +
-                                          c[property_index][5] * relative_support_point_location[1] * relative_support_point_location[1];
-                  }
-                else
-                  {
-                    interpolated_value += c[property_index][3] * relative_support_point_location[2] +
-                                          c[property_index][4] * relative_support_point_location[0] * relative_support_point_location[1] +
-                                          c[property_index][5] * relative_support_point_location[0] * relative_support_point_location[2] +
-                                          c[property_index][6] * relative_support_point_location[1] * relative_support_point_location[2] +
-                                          c[property_index][7] * relative_support_point_location[0] * relative_support_point_location[0] +
-                                          c[property_index][8] * relative_support_point_location[1] * relative_support_point_location[1] +
-                                          c[property_index][9] * relative_support_point_location[2] * relative_support_point_location[2];
-                  }
-
+                double interpolated_value = evaluate(c[property_index], relative_support_point_location);
                 // Overshoot and undershoot correction of interpolated particle property.
-                if (use_global_values_limiter)
+                if (use_quadratic_least_squares_limiter[property_index])
                   {
-                    interpolated_value = std::min(interpolated_value, global_maximum_particle_properties[property_index]);
-                    interpolated_value = std::max(interpolated_value, global_minimum_particle_properties[property_index]);
+                    interpolated_value = std::min(interpolated_value, property_maximums[property_index]);
+                    interpolated_value = std::max(interpolated_value, property_minimums[property_index]);
```

Comment:
This should not happen anymore after line 639, correct? For the bilinear least squares interpolator we added Asserts here that ensure we never exceed the bounds of the particle values after interpolation, can we do the same here?